specification.md

```diff
@@ -0,0 +1,501 @@
+    * It could be **dense**, in which case there must exist some kind of a mapping between the string labels and the integer indices.
+    * It could be **sparse**, in which case the labels could just be the string dimensions.
+* The auxiliary dataframes and arrays can be anything (e.g., `reducedDims` in SingleCellExperiment). These should be able to be efficiently joined with `obs` and `var` to produce the labels to query `X` with.
+* There can be more than one `X` arrays, and we will distinguish between two cases:
+    * They _share_ the same labels/coordinates and non-empty cells. In that case, we will talk about “layers” and use an intuitive API (see “attributes” in the TileDB implementation).
+    * They _do not share_ the same labels/coordinates. In that case, we will talk about separate arrays with their own coordinates.
+
+The following image summarizes the model for an **sc_group**, in the case where a set of `X` array layers shares the `obs` and `var` dataframes. This can be generalized to _multiple different groups_ of what is depicted below to capture cases where the  `X` array layers do not share the `obs` and `var` dataframes.
+
+![matrix schema](./images/matrix.png)
+
+And for the entire **sc_dataset**, containing one or more named **sc_group**:
+
+```
+sc_dataset                     # sc_dataset
+    |_ sc_group_1              # sc_group (contents as in the figure)
+    |_ sc_group_2              # sc_group (contents as in the figure)
+    |_ ...
+```
+
+In the most basic case, we need three queries:
+
+* The first uses any available dataframe and issues any query (even a multi-way join if needed), which involves the `obs` dataframe and produces a set of **row labels**, let’s call them  `obs_labels`.
+* The second uses any available dataframe and issues any query (even a multi-way join if needed), which involves the `var` dataframe and produces a set of **column labels**, let’s call them  `var_labels`.
+* Then the query will take as input an `X` array and potentially one of its “layers” (let’s denote that by `X[layer]`), `obs_labels` and `var_labels`.
+* The result of this query `Q(X[layer], obs_labels, var_labels)` can be represented in two forms:
+    * **Dense**, e.g., a 2D numpy array
+    * **Sparse**, e.g.,
+        * 3 1D numpy arrays, one for the row indices, one of the column indices, and one for the `X` values, or
+        * A pandas dataframe indexed on row and column indices for fast search
+
+## Exemplary Use Cases
+
+A handful of simple use cases to demonstrate possible usage.
+
+### Cloud Extraction to Scanpy/AnnData
+
+Unified single cell data corpus available on the cloud, at s3://giant-dataset/.  User's can open, slice and extract data in their format of choice.  In this case, we are extracting it as H5AD/AnnData. A similar method should be available for BioConductor & Seurat toolchains.
+
+```
+    import tiledb.sc as sc
```

Comment:
the comment in the gdoc says "changed 'sc' to 'api' in the markdown", but it is not changed in this example code.